### ISSUE TITLE
Update ossfuzz base image to newer Ubuntu (22.04 or 24.04) ?

### ISSUE THREAD

**Issue by rouault** [google/oss-fuzz#11835]
The GDAL project builds against the latest Poppler git version, but they have just increased their baseline dependencies (for boost, libtiff, freetype) to the ones provided by Ubuntu 22.04. So this requires also the GDAL ossfuzz build recipee to build those components (at least freetype which is the only one strictly required) from source, which complicates things. It could be worth considering updating to  ossfuzz base image to a newer Ubuntu

**Comment by oliverchang** [comment]
Yes, this is something we've been wanting to do, but as OSS-Fuzz has grown to the size it has over the years, it's also become a lot harder to do. 

One thing we're considering is to just pin all existing project images to the current Ubuntu 20.04, provide images based on newer Ubuntu, and let projects that do want to upgrade do the upgrade themselves.

**Comment by maxammann** [comment]
I know that this issue is kind of re-occurring since the release of Ubuntu 22. I was discussing this with a colleague and while the situation for the OSS-fuzz project is quite challenging (need to upgrade most/all projects to the new base images), it seems way more approachable for ClusterFuzzLite.

Especially for ClusterFuzzLite you typically fuzz projects that don't have a stable release cycle like e.g. OpenSSL. So it is quite a challenge to expect ClusterFuzzLite users to support Ubuntu 20. Most of the users are projects that don't quality for OSS-fuzz. And there is really no reason why CFL users shouldn't use a base image based off any Ubuntu version.

So our idea was to design a "channel" feature. As a CFL user you can define which channel/release of Ubuntu you want to target. Fundamentally this could also work with oss-fuzz in general but I don't have the full picture there.

**Comment by maxammann** [comment]
> This is unrelated to the Ubuntu version, as the compilers are self-compiled by oss-fuzz. Moving to a current rust nightly can be done as soon as most projects are un-pinned again. (If you want to help, they can be found via `git grep '@sha256:' ./projects/`)

Sadly not exactly. From Rust 1.77 top 1.78 (as far as I remember) they switched from LLVM 17 to 18 which makes coverage generated incompatible with the LLVM version used in the coverage container image.

So you get an error like `raw profile version mismatch: Profile uses raw profile format version = 9; expected version = 8` if you use the latest Rust nightly. Maybe I missed something, though

**Comment by evverx** [comment]
Now that https://github.com/google/oss-fuzz/pull/13964 is merged it seems Ubuntu-24.04 images are already available (and they can be used by adding the "ubuntu-24-04" tag). I'm not sure what the "latest" tag is going to point to (https://github.com/google/oss-fuzz/pull/14112) but currently it appears to be Ubuntu-20.04.

**Comment by pitrou** [comment]
> Now that [#13964](https://github.com/google/oss-fuzz/pull/13964) is merged it seems Ubuntu-24.04 images are already available (and they can be used by adding the "ubuntu-24-04" tag)

That works for building the fuzzers, the problem is that `infra/helper.py check_build` uses the default tag for the `base-runner` image and then fails to validate the fuzz targets on Ubuntu 20.04:

```
BAD BUILD: /tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz seems to have either startup crash or exit:
vm.mmap_rnd_bits = 28
/tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz -- -rss_limit_mb=2560 -timeout=25 -seed=1337 -runs=4 < /dev/null
/tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz)
/tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz)
/tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz)
/tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/not-out/tmpgw73ajsr/arrow-ipc-file-fuzz)
```

**Comment by evverx** [comment]
It was brought up in https://github.com/google/oss-fuzz/pull/14112#discussion_r2422427470 and the answer was
> For now, we recommend sticking with the current default version. In the coming weeks, we'll officially enable the option to select ubuntu-24-04 in project.yaml. By then, all the tooling will be ready to handle versioning intelligently.

The "gcr.io/oss-fuzz-base/base-runner:ubuntu-24-04" image is available though so with some patches it can be experimented with locally. Based on my experiments some 32-bit builds are broken on Ubuntu 24.04 and things like that.
